### PR TITLE
fix: spelling in the exercise

### PR DESCRIPTION
**Pull request by Tristan22400**
Fix: https://github.com/morpho-labs/hack-smart-contracts/issues/1

### CODE REVIEW

**Comment by makcandrov** [comment]
Exercice => Exercise (everywhere)

File: src/Hack.sol [hunk 1]
```diff
@@ -631,38 +644,52 @@ contract WinnerTakesAll {
...
         payable(msg.sender).transfer(amount);
     }
 
+    /// @dev Delete all the rounds created.
     function clearRounds() external onlyOwner {
         delete rounds;
     }
 
+    /// @dev WithDraws all the ethers to owner's address.
     function withrawETH() external onlyOwner {
```
Comment (makcandrov) [inline src/Hack.sol:693]:
```suggestion
    /// @dev Withdraws all the ethers to owner's address.
    function withdrawETH() external onlyOwner {
```

File: src/Hack.sol [hunk 2]
```diff
@@ -631,38 +644,52 @@ contract WinnerTakesAll {
...
+    /// @param _roundIndex The index of the round concerned by the reward.
     function setRewardsAtRound(uint256 _roundIndex) external payable onlyOwner {
         require(rounds[_roundIndex].rewards == 0);
         rounds[_roundIndex].rewards = msg.value;
     }
 
+    /// @dev Allows the participation of a set of addresses for a specific round.
+    /// @param _roundIndex The index of the round concerned.
+    /// @param _recipients The set of addresses allowed to participate.
     function setRewardsAtRoundfor(uint256 _roundIndex, address[] calldata _recipients) external onlyOwner {
```
Comment (makcandrov) [inline src/Hack.sol:665]:
```suggestion
    function setRewardsAtRoundFor(uint256 _roundIndex, address[] calldata _recipients) external onlyOwner {
```

File: src/Hack.sol [hunk 3]
```diff
@@ -631,38 +644,52 @@ contract WinnerTakesAll {
...
         for (uint256 i; i < _recipients.length; i++) {
             rounds[_roundIndex].isAllowed[_recipients[i]] = true;
         }
     }
 
+    /// @dev Checks if an address can participated to this round.
+    /// @param _roundIndex The index of the round concerned.
+    /// @param _recipient The address whose authorisation is to be checked.
```
Comment (makcandrov) [inline src/Hack.sol:673]:
```suggestion
    /// @dev Checks if an address can participate in this round.
    /// @param _roundIndex The index of the round to be checked.
    /// @param _recipient The address whose authorization is to be checked.
```

**Review by makcandrov — CHANGES_REQUESTED** [review]

File: src/Hack.sol [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
 pragma solidity ^0.8.0;
 
-// These contracts are examples of contracts with bugs and vulnerabilities in order to practice your hacking skills.
+// These contracts are examples of contracts with bugs and vulnerabilities to practice your hacking skills.
 // DO NOT USE THEM OR GET INSPIRATION FROM THEM TO MAKE CODE USED IN PRODUCTION
 // You are required to find vulnerabilities where an attacker harms someone else.
 // Being able to destroy your own stuff is not a vulnerability and should be dealt at the interface level.
 
-/// Exercice 1 ///
+/// Exercise 1 ///
```
Comment (MerlinEgalite) [inline src/Hack.sol:9]:
```suggestion
/* Exercise 1 */
```

and everywhere else

**Review by MerlinEgalite — APPROVED** [review]

**Review by MerlinEgalite — APPROVED** [review]

File: src/Hack.sol [hunk 5]
```diff
@@ -233,14 +239,14 @@ contract Coffers {
...
-    /// @param _slot The slot to deposit money.
+    /// @param _slot The slot to deposit money on.
     function deposit(address _owner, uint256 _slot) external payable {
         Coffer storage coffer = coffers[_owner];
         require(_slot < coffer.nbSlots);
         coffer.slots[_slot] += msg.value;
     }
 
-    /// @dev Withdraws all of the money of one's coffer slot.
+    /// @dev Withdraws all of the money from one's coffer slot.
```
Comment (makcandrov) [inline src/Hack.sol:249]:
```suggestion
    /// @dev Withdraws all the money from one's coffer slot.
```

File: src/Hack.sol [hunk 6]
```diff
@@ -233,14 +239,14 @@ contract Coffers {
 
     /// @dev Deposits money in one's coffer slot.
```
Comment (makcandrov) [inline src/Hack.sol:240]:
```suggestion
    /// @dev Deposits money into one's coffer slot.
```

File: src/Hack.sol [hunk 7]
```diff
@@ -233,14 +239,14 @@ contract Coffers {
...
     /// @param _owner The coffer to deposit money on.
```
Comment (makcandrov) [inline src/Hack.sol:241]:
```suggestion
    /// @param _owner The owner of the coffer.
```

File: src/Hack.sol [hunk 8]
```diff
@@ -233,14 +239,14 @@ contract Coffers {
...
-    /// @param _slot The slot to deposit money.
+    /// @param _slot The slot to deposit money on.
```
Comment (makcandrov) [inline src/Hack.sol:242]:
```suggestion
    /// @param _slot The slot to deposit money into.
```

**Review by makcandrov — APPROVED** [review]

**Pull request merged**